circuits/cpp/src/aztec3/circuits/abis/signed_tx_request.hpp

```diff
@@ -51,10 +46,40 @@ template <typename NCT> struct SignedTxRequest {
 
         SignedTxRequest<NativeTypes> signed_tx_request = {
             to_native_type(tx_request),
+            to_native_type(signature),
         };
 
         return signed_tx_request;
     };
+
+    fr compute_signing_message() const
+    {
+        std::vector<fr> messages;
+        messages.push_back(tx_request.from.to_field());
+        messages.push_back(tx_request.to.to_field());
+        messages.push_back(tx_request.function_data.hash());
+        messages.push_back(NCT::compress(tx_request.args, aztec3::CONSTRUCTOR_ARGS));
+        messages.push_back(tx_request.nonce);
+        messages.push_back(tx_request.tx_context.hash());
+        messages.push_back(tx_request.chain_id);
+
+        return NCT::compress(messages, aztec3::SIGN_TX_REQUEST);
+    }
+
+    void compute_signature(const NativeTypes::secp256k1_fr& private_key)
+    {
+        static_assert((std::is_same<NativeTypes, NCT>::value));
+        auto signing_message = compute_signing_message().to_buffer();
+        std::string signing_message_str(signing_message.begin(), signing_message.end());
+        crypto::ecdsa::key_pair<NativeTypes::secp256k1_fr, NativeTypes::secp256k1_group> account;
+        account.private_key = private_key;
+        account.public_key = tx_request.from_public_key;
+
+        signature = crypto::ecdsa::construct_signature<Sha256Hasher,
+                                                       NativeTypes::secp256k1_group::Fq,
+                                                       NativeTypes::secp256k1_group::Fr,
+                                                       NativeTypes::secp256k1_group>(signing_message_str, account);
+    }
```

Comment:
Will the typescript call this method (via a c_bind)? Or are we using some ethereum library to compute an ethereal signature in typescript?

Reply:
Added a comment here. So I am not using this function in TS for the following reason: the way TS computes the signature is:

message = hashTxRequest(txRequest)
signature = keyPair.signMessage(message)
So we don't really need to expose this function as a c_bind. I am only using this function to construct a signature for testing in C++.